packages/gatsby/src/redux/actions/internal.ts

```diff
@@ -350,3 +357,58 @@ export const deleteNodeManifests = (): IDeleteNodeManifests => {
     type: `DELETE_NODE_MANIFESTS`,
   }
 }
+
+export const createJobV2FromInternalJob = (
+  internalJob: InternalJob
+): ICreateJobV2FromInternalAction => (
+  dispatch,
+  getState
+): Promise<Record<string, unknown>> => {
+  const jobContentDigest = internalJob.contentDigest
+  const currentState = getState()
+
+  // Check if we already ran this job before, if yes we return the result
+  // We have an inflight (in progress) queue inside the jobs manager to make sure
+  // we don't waste resources twice during the process
+  if (
+    currentState.jobsV2 &&
+    currentState.jobsV2.complete.has(jobContentDigest)
+  ) {
+    return Promise.resolve(
+      // @ts-ignore we check if `complete` map has result
```

Comment:
What does that mean exactly?

Reply:
TS was complaining that next line's `.get` might return `something | undefined` even tho we did check with `.has` above (so it shouldn't have `undefined` as possible type)

Reply:
Ok, yeah, reading https://github.com/microsoft/TypeScript/issues/9619 I see it now. Alternatively you could also add a `!` to say that it'll be defined. I guess that's preferable compared to a ts-ignore

Reply:
+1 to `!.`

Reply:
dropped `// @ts-ignore` in https://github.com/gatsbyjs/gatsby/pull/32054/commits/68fc0889472bb1cef540ec850cdebc73ff5ade53 as I already had `!` in there - my IDE was still not liking it, but typecheck and eslint seems to pass